### ISSUE TITLE
Test case: mv_files_002_pos

### ISSUE THREAD

**Issue by behlendorf** [zfsonlinux/zfs#7220]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | CentOS
Distribution Version    | 7
Linux Kernel                 | 4.8.1-1.el7.elrepo.x86_64
Architecture                 | x86_64
ZFS Version                  | zfs-0.7.0-334-g7088545
SPL Version                  | spl-0.7.0-29-g378c6ed

### Describe the problem you're observing

Occasional failure of `mv_files_002_pos` test case.

### Describe how to reproduce the problem

Occasionally reproduced by buildbot during automated testing.

### Include any warning/errors/backtraces from the system logs

http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20Mainline%20%28TEST%29/builds/5643

```
Test: /usr/share/zfs/zfs-tests/tests/functional/mv_files/mv_files_002_pos (run as root) [01:08] [FAIL]
17:40:14.33 ASSERTION: Doing a 'mv' of a large amount of files across two zfs filesystems works without errors.
17:41:22.17 The file number of target directory 3000 is not equal to that of the source  directory /mnt/testdir_tgt/newdir
```

**Comment by behlendorf** [comment]
Reopening.  This appears not to have been completely addressed by d0f6fbaff383d0ea0b11f4e4547d6b062ef7ddad, a failure was observed on master with this patch applied.  @chrisrd any thoughts?

http://build.zfsonlinux.org/builders/CentOS%207%20x86_64%20%28TEST%29/builds/2107/steps/shell_8/logs/summary

```
Test: /usr/share/zfs/zfs-tests/tests/functional/mv_files/mv_files_002_pos (run as root) [00:28] [FAIL]
01:47:42.44 ASSERTION: Doing a 'mv' of a large amount of files across two zfs filesystems works without errors.
01:48:09.90 The file number of target directory 3000 is not equal to that of the source  directory /mnt/testdir_tgt/newdir
```

**Comment by behlendorf** [comment]
Closing, resolved by 4f301661dfe7e90574ad634f8b7fe8da7e6b4064.
